src/analysis/bounds.rs

```diff
@@ -79,61 +96,117 @@ impl Bounds {
+                Type::SList(_) |
+                Type::CArray(_) => None,
+            Type::Fundamental(_) if *nullable => Some(Into(None, None)),
+            _ if !*nullable => None,
+            _ => Some(Into(Some('_'), None)),
         }
     }
     fn to_glib_extra(bound_type: BoundType) -> String {
         use self::BoundType::*;
         match bound_type {
-            AsRef => ".as_ref()".to_owned(),
-            Into(_) => ".into()".to_owned(),
+            AsRef(_) => ".as_ref()".to_owned(),
+            //Into(_, None) => ".into()".to_owned(),
+            Into(_, Some(x)) => format!("{}", Bounds::to_glib_extra(*x)),
             _ => String::new(),
         }
     }
     pub fn add_parameter(&mut self, name: &str, type_str: &str, mut bound_type: BoundType) -> bool {
-        if self.used.iter().any(|n| n.parameter_name == name) { return false; }
-        if let BoundType::Into(_) = bound_type {
+        if self.used.iter().any(|ref n| n.parameter_name == name) { return false; }
+        let sub = if let BoundType::Into(Some(_), x) = bound_type {
             if let Some(lifetime) = self.unused_lifetimes.pop_front() {
                 self.lifetimes.push(lifetime);
-                bound_type = BoundType::Into(lifetime);
+                bound_type = BoundType::Into(Some(lifetime), x.clone());
+                Some((x, lifetime))
             } else {
                 return false;
             }
-        }
+        } else {
+            None
+        };
+        let type_str = if let Some((Some(sub), lifetime)) = sub {
+            if let Some(alias) = self.unused.pop_front() {
+                self.used.push(Bound {
+                    bound_type: BoundType::set_lifetime(*sub, lifetime),
+                    parameter_name: name.to_owned(),
+                    alias: alias,
+                    type_str: type_str.to_owned(),
+                    info_for_next_type: true,
+                });
+                format!("{}", alias)
+            } else {
+                return false;
+            }
+        } else {
+            type_str.to_owned()
+        };
         if let Some(alias) = self.unused.pop_front() {
-            self.used.push(Bound{
+            self.used.push(Bound {
                 bound_type: bound_type,
                 parameter_name: name.to_owned(),
                 alias: alias,
                 type_str: type_str.to_owned(),
+                info_for_next_type: false,
             });
             true
         } else {
             false
         }
     }
     pub fn get_parameter_alias_info(&self, name: &str) -> Option<(char, BoundType)> {
-        self.used.iter().find(|n| n.parameter_name == name)
-            .map(|t| (t.alias, t.bound_type))
+        let mut stop = false;
+        self.used.iter()
+                 .find(move |ref n| {
+            if stop == true {
+                true
+            } else if n.parameter_name == name {
+                if n.info_for_next_type == false {
```

Comment:
can changed just to `!n.info_for_next_type` with same result, `stop` unneeded then